hnix-store-binary/src/System/Nix/Store/Binary.hs

```diff
@@ -0,0 +1,93 @@
+{-|
+Description : Interact with a binary Nix store.
+Maintainer  : Brian McKenna <brian@brianmckenna.org>
+-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module System.Nix.Store.Binary (
+  BinaryStoreEffects(..)
+, addToStore
+) where
+
+import Codec.Compression.Lzma (compress)
+import Control.Monad.Trans (MonadIO(..))
+import Data.Text (Text)
+import Data.Time.Clock.POSIX (posixSecondsToUTCTime)
+import System.Nix.Hash
+import System.Nix.Nar
+import System.Nix.StorePath
+import System.Nix.StorePathMetadata
+import qualified Data.Binary.Put as B
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.HashSet as HS
+import qualified Data.Text.Encoding as E
+import System.Nix.ReadonlyStore
+
+data BinaryStoreEffects m =
+  BinaryStoreEffects
+    { upsertFile :: Text -> BS.ByteString -> Text -> m ()
+    }
+
+narInfoFileFor
+  :: StorePath storeDir
+  -> Text
+narInfoFileFor p =
+  encodeBase32 (storePathHash p) <> ".narinfo"
```

Comment:
This is now part of `-core`s `System.Nix.StorePath` called `storePathToNarInfo`.